internal/controllers/vgmanager/controller.go

```diff
@@ -689,6 +704,26 @@ func (r *Reconciler) extendThinPool(ctx context.Context, vgName string, lvSize s
 	return nil
 }
 
+func (r *Reconciler) verifyMetadataSize(ctx context.Context, vgName, lvName string, lvMetadataSize string, newMetadataSize int64) error {
+	if newMetadataSize == -1 {
+		return nil
+	}
+
+	currentMetadataSize, err := strconv.ParseInt(lvMetadataSize[:len(lvMetadataSize)-1], 10, 64)
+	if err != nil {
+		return fmt.Errorf("failed to parse metadata size. %w", err)
+	}
+	if currentMetadataSize >= newMetadataSize {
+		return nil
+	}
+
+	err = r.LVM.ExtendThinPoolMetadata(ctx, lvName, vgName, newMetadataSize)
```

Comment:
I believe this method should only do the validation as it is part of `validateLVs`. The actual extend operation can take place within `extendThinPool` called [here](https://github.com/openshift/lvm-operator/blob/3ca641782ca8ccfd7637d8c385ace913884459e7/internal/controllers/vgmanager/controller.go#L611) or just after it as they are closely related.